symfony/plugins/orangehrmCorePlugin/Service/EmailQueueService.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Service;
+
+use OrangeHRM\Core\Dao\EmailQueueDao;
+use OrangeHRM\Entity\Mail;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface as MailerException;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Core\Exception\CoreServiceException;
+use DateTime;
+
+class EmailQueueService
+{
+    private ?EmailQueueDao $emailQueueDao = null;
+
+    private ?EmailService $emailService = null;
+
+    /**
+     * @return EmailQueueDao|null
+     */
+    public function getEmailQueueDao(): EmailQueueDao
+    {
+        if (!($this->emailQueueDao instanceof EmailQueueDao)) {
+            $this->emailQueueDao = new EmailQueueDao();
+        }
+        return $this->emailQueueDao;
+    }
+
+    /**
+     * @return EmailService
+     * @throws DaoException
+     * @throws CoreServiceException
+     */
+    public function getEmailService(): EmailService
+    {
+        if (!($this->emailService instanceof EmailService)) {
+            $this->emailService = new EmailService();
+            $this->loadConfiguration();
+        }
+        return $this->emailService;
+    }
+
+    /**
+     * @throws CoreServiceException
+     * @throws DaoException
+     */
+    public function loadConfiguration(): void
+    {
+        $this->getEmailService()->loadConfiguration();
```

Comment:
isn't it good to move this to `EmailService` constructor?

Reply:
moved